### DISCUSSION TITLE
Book for Beginner?

### DISCUSSION THREAD

**Discussion by matthornish** [nextgenhealthcare/connect#5030] [Q&A] [answered]
Hello, everyone.  I have some experience with MIRTH, but it is mainly with setting up sources and destinations to route messages.  I am looking to delve a little more into transforming messages to better adapt to systems that require a bit more heavy lifting.  Can anyone recommend a good book to read along those lines?

Thanks in advance!

**Comment by pacmano1** [comment, marked as answer]
Do read https://github.com/nextgenhealthcare/connect/discussions/4659.  

Also see http://mirthconnect.shamilpublishing.com/, I have heard good things about it.

And https://www.amazon.com/Up-Speed-Mirth-Connect-NextGen/dp/B08JF5FGTG/ref=sr_1_3?keywords=up+to+speed+with+mirth+connect&qid=1645131523&sr=8-3

If you can engage a senior consultant though that will go a long way.

**Reply by matthornish** [reply]
Thanks!  I'll take a look!

**Reply by muhammadc** [reply]
**Fun fact**: the author of "Up to Speed with Mirth Connect" is now one of the official Mirth Connect trainers at NextGen.

**Reply by matthornish** [reply]
Seems like a solid endorsement. Thanks!

**Comment by jonbartels** [comment]
Reading discussions and https://forums.mirthproject.io/ or joining Slack https://mirthconnect.herokuapp.com/ can help you see real-world problems that others are solving and learn from them.

**Reply by matthornish** [reply]
Thanks!
